safetensors/src/tensor.rs

```diff
@@ -507,7 +517,7 @@ impl Metadata {
             tensors,
             index_map,
         };
-        // metadata.validate()?;
+
```

Comment:
While working on a secondary crate, I realized there was no public API for Metadata, so I marked this as pub in another PR.

Since it's now public we really need to keep validate (so I uncommented it), to make sure the Metadata can only be valid. (Distinc names, ordered data_offsets vectors etc..)

Reply:
This doesn't remove the function itself, only the call. So are you saying the call itself is needed? (Otherwise, I would see no reason to keep around commented-out code.)

Reply:
Yes, it was commented out as it was only an internal function, and at some point it was easier to remove the call on specific pathways to allow reading "malformed"  data (Metadata can follow the spec, but still not respect Metadata layout, which would cause issues down the line). While it wasn't public, we made sure the data was valid in other ways, since it's public we must reject any invalid data directly.